R/utilities-building-block.R

```diff
@@ -0,0 +1,177 @@
+#'
+#' @param pvBuildingBlock A `BuildingBlock` object of type `Parameter Values`.
+#' The entries will be added to or set in this building block.
+#' @param quantityPaths A list of full paths of the quantities (usually parameters). Should contain
+#' all path elements and the parameter name, separated by `|`.
+#' @param formula A list of `Formula` objects that will be set for the quantities.
+#' Mixing of `formula` and `quantityValues` is not allowed. If `formula` is provided,
+#' `quantityValues` should be `NULL`. The length of this list should be equal to the length of `quantityPaths`.
+#' @param dimensions A single dimension or a list of dimensions (string names)
+#' of parameter values. Supported dimensions are listed in `ospDimension`. By default,
+#' new entries get the `Dimensionless` dimension.
+#' @param quantityValues A list of values for the quantities. The length of this list should be equal to the length of `quantityPaths`.
+#'
+#' @export
+#'
+#' @examples
+setParameterValues <- function(pvBuildingBlock,
+                                 quantityPaths,
+                               quantityValues,
+                               dimensions = ospDimensions$Dimensionless,
+                                 formula = NULL){
+
+  return(pvBuildingBlock)
+}
+
+#' Delete entries from a Parameter Values Building Block
+#'
+#' @param pvBuildingBlock A `BuildingBlock` object of type `Parameter Values`.
+#' @param quantityPaths A list of full paths of the quantities (usually parameters)
+#' that will be deleted. Should contain all path elements and the parameter name, separated by `|`.
+#' Entries not present in the provided BB are ignored.
+#'
+#' @export
+#'
+#' @examples
+deleteParameterValues <- function(pvBuildingBlock,
+                                    quantityPaths){
+}
+
+#' Extend a Parameter Values Building Block (BB) with local molecule parameters
+#' for molecules from a molecules BB in all physical containers of a spatial structure BB.
+#'
+#' @param pvBuildingBlock A `BuildingBlock` object of type `Parameter Values`.
+#' @param spatialStructureBB A `BuildingBlock` object of type `Spatial Structure`.
+#' Entries will be created for local parameters of the selected molecules in all physical containers of this
+#' spatial structure.
+#' @param moleculesBB A `BuildingBlock` object of type `Molecules`. The entries will be
+#' created for all molecules from this building block, or for a subset of molecules
+#' defined in the `moleculeNames` argument.
+#' @param moleculeNames Optional list of molecule names. If provided, only the molecules
+#' with these names will be added to the `pvBuildingBlock`.
+#'
+#' @returns Path of entries added to the building block.
+#' @export
+#'
+#' @examples
+addLocalMoleculeParameters <- function(pvBuildingBlock,
+                                    spatialStructureBB,
+                                    moleculesBB,
+                                    moleculeNames = NULL){
```

Comment:
same question/suggestion as for the initial conditions (optional list of container start paths)